packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp

```diff
@@ -51,18 +54,22 @@
 #include <list>
 #include "ApiTest.h"
 
+
+
 class DeepCopyTester {
 public:
   DeepCopyTester() {}
-  void run() {
+  void run(int argc, char *argv[])  {
     log("Tpetra Kokkos DeepCopy Regression test start");
 
     std::string cudaSync("cudaDeviceSynchronize");
     std::string cudaMemcpy("cudaMemcpy");
     std::string cudaMemcpyAsync("cudaMemcpyAsync");
     ApiTest *counter = ApiTest::getInstance();
     //initialize
+#ifdef HAVE_TPETRACORE_MPI
     MPI_Init(&argc, &argv);
```

Comment:
don't we have a Tpetra::ScopeGuard that can take care of some of this logic for us?  But also, shouldn't this initialize/finalize stuff get called in main instead?

Reply:
Yeah, all of this initialize / finalize is an artifact of the way this was written.